docs/sdk/src/reference_docs/frame_pallet_coupling.rs

```diff
@@ -0,0 +1,262 @@
+//! via an associated type:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_config)]
+//!
+//! Then, `pallet-foo` can use this trait to obtain the block author, without knowing where it comes
+//! from:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_usage)]
+//!
+//! Then, if `pallet-author` implements this glue-trait:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author_provider)]
+//!
+//! And upon the creation of the runtime, the two pallets are linked together as such:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", runtime_author_provider)]
+//!
+//! Crucially, in when using loose coupling, we gain the flexibility of providing different
+//! implementations of `AuthorProvider`, such that different users of a `pallet-foo` can use
+//! different ones, without any code change being needed. For example, in the code snippets of this
+//! module, you can fund [`OtherAuthorProvider`] which is an alternative implementation of
+//! [`AuthorProvider`].
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", other_author_provider)]
+//!
+//! ## Frame System
+//!
+//! With the above information in context, we can conclude that `frame_system` is a special pallet
+//! that is tightly coupled with every other pallet. This is because it provides the fundamental
+//! system functionality that every pallet needs, such as some types like `AccountId`, `Hash`, and
+//! some functionality such as block number, etc.
+//!
+//! ## Recap
+//!
+//! To recap, consider the following rules of thumb:
+//!
+//! * In all cases, try and break down big pallets apart with clear boundaries of responsibility. In
+//!   general, it is easier to argue about multiple pallet if they only communicate together via a
+//!   known trait, rather than having access to all of each others public items, such as storage and
+//!   dispatchables.
+//! * If a group of pallets are meant to work together, and but are not foreseen to be generalized,
+//!   or used by others, consider tightly coupling pallets, *if it simplifies the development*.
+//! * If a pallet needs a functionality provided by another pallet, but multiple implementations can
+//!   be foreseen, consider loosely coupling pallets.
+//!
+//! For example, all pallets in `polkadot-sdk` that needed to work with currencies could have been
+//! tightly coupled with [`pallet_balances`]. But, `polkadot-sdk` also provides [`pallet_assets`]
+//! (and more implementations by the community), therefore all pallets use traits to loosely couple
+//! with balances or assets pallet. More on this in [`crate::reference_docs::frame_currency`].
+//!
+//! ## Further References
+//!
+//! <https://www.youtube.com/watch?v=0eNGZpNkJk4>
+//! <https://substrate.stackexchange.com/questions/922/pallet-loose-couplingtight-coupling-and-missing-traits>
```

Comment:
Otherwise it renders both items on the same line
```suggestion
//! - <https://www.youtube.com/watch?v=0eNGZpNkJk4>
//! - <https://substrate.stackexchange.com/questions/922/pallet-loose-couplingtight-coupling-and-missing-traits>
```